examples/modals/uploader.py

```diff
@@ -0,0 +1,111 @@
+        await self.tree.sync(guild=TEST_GUILD)
+
+
+class BotUpdateModal(discord.ui.Modal, title='Set Bot Avatar & Banner'):
+    # We can use a Label to attach a rich label and description to our item.
+    avatar = discord.ui.Label(
+        text='Avatar',
+        description='The new avatar for the bot. Optional.',
+        component=discord.ui.FileUpload(
+            max_values=1,  # The maximum number of files that can be uploaded
+            required=False,
+        ),
+    )
+
+    banner = discord.ui.Label(
+        text='Banner',
+        description='The new banner for the bot. Optional.',
+        component=discord.ui.FileUpload(
+            max_values=1,
+            required=False,
+        ),
+    )
+
+    async def on_submit(self, interaction: discord.Interaction[MyClient]):
+        # We defer the response because updating the avatar and banner can take a bit of time
+        await interaction.response.defer()
+
+        # Get our bot to update its avatar and banner
+        client = interaction.client
+
+        # Tell the type checker what our components are...
+        assert isinstance(self.avatar.component, discord.ui.FileUpload)
+        assert isinstance(self.banner.component, discord.ui.FileUpload)
+
+        avatar: Optional[discord.Attachment] = self.avatar.component.values[0] if self.avatar.component.values else None
+        banner: Optional[discord.Attachment] = self.banner.component.values[0] if self.banner.component.values else None
+
+        # Update the avatar and banner
+        # We read the bytes from the attachment and pass it to edit
+        # If no file was uploaded, we pass `discord.utils.MISSING`, to not update that field
+        await client.user.edit(
+            avatar=await avatar.read() if avatar else discord.utils.MISSING,
+            banner=await banner.read() if banner else discord.utils.MISSING,
+        )
+
+        # Let the user know it was successful, this is optional
+        await interaction.followup.send(
+            f"Updated {client.user.mention}'s avatar / banner.",
+            ephemeral=True,
+        )
+
+    async def on_error(self, interaction: discord.Interaction, error: Exception) -> None:
+        await interaction.response.send_message('Oops! Something went wrong.', ephemeral=True)
+
+        # Make sure we know what the error actually is
+        traceback.print_exception(type(error), error, error.__traceback__)
+
+
+client = MyClient()
+
+
+@client.tree.command(guild=TEST_GUILD, description="Update the bot's avatar and banner")
+async def update_bot(interaction: discord.Interaction):
+    # Locking this command to the bot owner only
+    # We don't want anyone else changing our bot's avatar and banner!
+    # Realistically, this should be proper checks, but this is just an example.
+    if interaction.user.id != BOT_OWNER_ID:
+        await interaction.response.send_message('You are not the owner of this bot.', ephemeral=True)
+        return
```

Comment:
We should practice what we preach in the discord server. We should not be making "public access" commands that are restricted to an owner.

Could we change the example command to something like anonymous photo submission or similar?